webapp_v2/src/common/types.ts

```diff
@@ -586,4 +586,39 @@ export interface TaskView {
   status: string;
 }
 
+export interface XpansionSettings {
+  uc_type: string;
+  master: string;
+  optimality_gap: number;
+  max_iteration?: number;
+  "yearly-weights": string;
+  "additional-constraints"?: string;
+  "relaxed-optimality-gap"?: number;
+  cut_type?: string;
+  "ampl.solver"?: string;
+  "ampl.presolve"?: number;
+  "ampl.solve_bounds_frequency"?: number;
+  relative_gap?: number;
+  solver?: string;
+}
+
+export interface XpansionCandidate {
+  name: string;
+  link: string;
+  "annual-cost-per-mw": number | null;
+  "unit-size"?: number;
+  "max-units"?: number;
+  "max-investment"?: number;
+  "already-installed-capacity"?: number;
+  "link-profile"?: string | null;
+  "already-installed-link-profile"?: string | null;
+}
+
+export enum XpansionRenderView {
+  candidate = "candidate",
+  settings = "settings",
+  files = "files",
+  capacities = "capacities",
+}
+
```

Comment:
you can put this in types.ts under components/.../Xpansion folder (they are not used anywhere else)